.github/workflows/ci.yml

```diff
@@ -63,3 +66,100 @@ jobs:
     - name: Build kube-router
       run: |
         make kube-router
+
+  # Builds Container on Push Event, not run if there's a Tag Event
+  ci-build-container-push:
+    runs-on: ubuntu-latest
+    if: ${{ startsWith('refs/tags/v', github.ref) != true && github.event_name != 'pull_request'  }}
+    steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        platforms: linux/amd64,linux/arm64,linux/arm,linux/ppc64,linux/s390x
+        push: true
+        tags: cloudnativelabs/kube-router:${{ github.sha }}
+
+  # Builds Container on Push Event, not run if there's a Tag Event
+  ci-build-container-pr:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        platforms: linux/amd64,linux/arm64,linux/arm,linux/ppc64,linux/s390x
+        push: true
+        tags: cloudnativelabs/kube-router:${{ github.sha }},cloudnativelabs/kube-router:PR-${{ github.event.pull_request.number }}
+
+  # Builds Container on Tag Event
+  ci-build-container-tag:
+    runs-on: ubuntu-latest
+    if: ${{ startsWith('refs/tags/v', github.ref) }}
```

Comment:
Do we need to add `&& github.event_name != 'pull_request'` here?

If someone named their branch `vfoobar` as part of a PR, would it match this? Or would GitHub not consider it a `ref` action, or use the target `ref` (i.e. `master`) here?

Reply:
It's checking refs/tags/ and not refs/heads, so it should not be an issue.